### PR TITLE
Re-implements image listing to sort/filter on jumpgate itself

### PR DESCRIPTION
**Pull request by sudorandom**
This should fix the issue with specifying a marker when listing images not working properly. This solution is not optimal since it sorts and filters client side.

### CODE REVIEW

File: jumpgate/image/drivers/sl/images.py [hunk 1]
```diff
@@ -385,43 +143,23 @@ def on_get(self, req, resp, tenant_id=None):
...
-                                                            req,
-                                                            image,
-                                                            tenant_id)
-
-                if formatted_image:
-                    formatted_image['visibility'] = visibility
-                    output.append(formatted_image)
-                    if limit is not None:
-                        limit -= 1
+            sorted_images = sorted_images[:int(req.get_param('limit'))]
```
Comment (jimlindeman) [inline jumpgate/image/drivers/sl/images.py:159]: What if limit was None here?  Now normally, does not make sense to pass marker parameter without a limit parameter but probably need a default here inside this array subset specification.  I.e.
sorted_images = sorted_images[:int(req.get_param('limit') or "20")]

Comment (sudorandom) [inline jumpgate/image/drivers/sl/images.py:159]: It's hard to read in this diff, but the list is not limited at all if limit isn't passed:

``` python
if req.get_param('limit'):
    sorted_images = sorted_images[:int(req.get_param('limit'))]
```

But yeah, having a default limit might be good

Comment (jimlindeman) [inline jumpgate/image/drivers/sl/images.py:159]: You're right, missed the hidden line 158 in all the diff-deletes.  This should be fine for now.

**Comment by jimlindeman** [comment]
+1
Looks good

**Pull request merged**